### PR TITLE
implement `replace_triggers_refs` for `azapi_resource`

### PR DESCRIPTION
**Pull request by hqhqhqhqhqhqhqhqhqhqhq**
for resource `azapi_resource`, the `replace_triggers_refs` keep a list of paths (JMES supported) pointing to values in the resource's **body**, when the value pointed by any one of the path is changed, the resource will be replaced.

example:
```
resource "azapi_resource" "example" {
  type      = "Microsoft.Relay/namespaces@2021-11-01"
  parent_id = azurerm_resource_group.example.id
  name      = "xxx"
  location  = "westus"
  body = {
    properties = {
    }
    sku = {
      name = "Standard"
      tier = "Standard"
    }
  }

  replace_triggers_refs = ["sku"]
}
```

The current `sku` value is 
```
{
  name = "Standard"
  tier = "Standard"
}
```
if there is any changes above, the whole resource will be replaced.

### CODE REVIEW

File: internal/services/azapi_resource.go [hunk 1]
```diff
@@ -196,6 +197,12 @@ func (r *AzapiResource) Schema(ctx context.Context, _ resource.SchemaRequest, re
 				},
 			},
 
+			"replace_triggers_refs": schema.ListAttribute{
```
Comment (ms-henglu) [inline internal/services/azapi_resource.go:200]: Please also implement this feature in the data plane resource.

Comment (ms-henglu) [inline internal/services/azapi_resource.go:200]: Please also implement this feature in the data plane resource.

File: internal/services/azapi_resource.go [hunk 2]
```diff
@@ -499,6 +506,36 @@ func (r *AzapiResource) ModifyPlan(ctx context.Context, request resource.ModifyP
...
+		err = json.Unmarshal([]byte(plan.Body.String()), &data)
+		if err != nil {
+			response.Diagnostics.AddError("Invalid configuration", err.Error())
+			return
+		}
+		currentValues := flattenOutputJMES(data, refPaths)
+
+		// compare previous and current values
+		if !reflect.DeepEqual(previousValues, currentValues) {
+			response.RequiresReplace.Append(path.Root("body"))
```
Comment (ms-henglu) [inline internal/services/azapi_resource.go:536]: Is it possible to add the specific path like "body.properties.sku.name" that triggers the replacement?

**Review by ms-henglu — CHANGES_REQUESTED** [review]
Thanks @hqhqhqhqhqhqhqhqhqhqhq for this PR! LGTM, just some minor suggestions, please check.

File: internal/services/azapi_resource.go (hunk 2 above)
Comment (hqhqhqhqhqhqhqhqhqhqhq) [reply internal/services/azapi_resource.go:536]: I tried this, the cli will say:
```
Provider "registry.terraform.io/azure/azapi" has indicated "requires replacement" on azapi_resource.example for a
│ non-existent attribute path cty.Path{cty.GetAttrStep{Name:"body.properties.sku.name"}}.
```

Similar error message if I try with `body.sku.name`

File: internal/services/azapi_resource.go (hunk 1 above)
Comment (hqhqhqhqhqhqhqhqhqhqhq) [reply internal/services/azapi_resource.go:200]: Got it, any need for other resources too?

Comment (ms-henglu) [reply internal/services/azapi_resource.go:200]: Thanks! No.

Comment (hqhqhqhqhqhqhqhqhqhqhq) [reply internal/services/azapi_resource.go:200]: Added & tested

**Review by ms-henglu — APPROVED** [review]

**Pull request merged**